diff --git a/dev/archery/archery/integration/datagen.py b/dev/archery/archery/integration/datagen.py
index 6a077a893..cab6ecd37 100644
--- a/dev/archery/archery/integration/datagen.py
+++ b/dev/archery/archery/integration/datagen.py
@@ -1561,8 +1561,7 @@ def get_generated_json_files(tempdir=None):
         .skip_category('C#')
         .skip_category('JS'),   # TODO(ARROW-7900)
 
-        generate_decimal128_case()
-        .skip_category('Rust'),
+        generate_decimal128_case(),
 
         generate_decimal256_case()
         .skip_category('Go')  # TODO(ARROW-7948): Decimal + Go
@@ -1574,18 +1573,15 @@ def get_generated_json_files(tempdir=None):
 
         generate_interval_case()
         .skip_category('C#')
-        .skip_category('JS')  # TODO(ARROW-5239): Intervals + JS
-        .skip_category('Rust'),
+        .skip_category('JS'),  # TODO(ARROW-5239): Intervals + JS
 
         generate_month_day_nano_interval_case()
         .skip_category('C#')
-        .skip_category('JS')
-        .skip_category('Rust'),
+        .skip_category('JS'),
 
 
         generate_map_case()
-        .skip_category('C#')
-        .skip_category('Rust'),
+        .skip_category('C#'),
 
         generate_non_canonical_map_case()
         .skip_category('C#')
@@ -1602,14 +1598,12 @@ def get_generated_json_files(tempdir=None):
         generate_nested_large_offsets_case()
         .skip_category('C#')
         .skip_category('Go')
-        .skip_category('JS')
-        .skip_category('Rust'),
+        .skip_category('JS'),
 
         generate_unions_case()
         .skip_category('C#')
         .skip_category('Go')
-        .skip_category('JS')
-        .skip_category('Rust'),
+        .skip_category('JS'),
 
         generate_custom_metadata_case()
         .skip_category('C#')
@@ -1634,14 +1628,12 @@ def get_generated_json_files(tempdir=None):
         .skip_category('C#')
         .skip_category('Go')
         .skip_category('Java')  # TODO(ARROW-7779)
-        .skip_category('JS')
-        .skip_category('Rust'),
+        .skip_category('JS'),
 
         generate_extension_case()
         .skip_category('C#')
         .skip_category('Go')  # TODO(ARROW-3039): requires dictionaries
-        .skip_category('JS')
-        .skip_category('Rust'),
+        .skip_category('JS'),
     ]
 
     generated_paths = []
